osidb/models/tracker.py

```diff
@@ -170,19 +168,11 @@ def save(self, *args, bz_api_key=None, jira_token=None, **kwargs):
                 self._validate_tracker_duplicate()
             # sync to Bugzilla
             tracker_instance = TrackerSaver(self, bz_api_key=bz_api_key).save()
-            # save in case a new Bugzilla ID was obtained
-            # so the tracker is later matched in BZ import
-            kwargs[
-                "auto_timestamps"
-            ] = False  # the timestamps will be get from Bugzilla
-            tracker_instance.save(*args, **kwargs)
-            # fetch from Bugzilla
-            btc = BugzillaTrackerCollector()
-            btc.no_alerts = True
-            btc.sync_tracker(tracker_instance.external_system_id)
-            BZTrackerLinkManager.link_tracker_with_affects(
-                tracker_instance.external_system_id
-            )
+            # no save or fetch to prevent collisions
+            # only schedule an asynchronous sync
+            BZTrackerDownloadManager.schedule(tracker_instance.external_system_id)
+            # at the end delete the temporary empty tacker
+            Tracker.objects.filter(external_system_id="").delete()
```

Comment:
I am just thinking, do we need to actually create the Tracker instance with empty `external_system_id` and store it in the database? TrackerSaver operates with the django model instance and it shouldn't matter whether it is saved or not, right? If that is the case, we could just not save the tracker [here in serializer](https://github.com/RedHatProductSecurity/osidb/blob/master/osidb/serializer.py#L702) and work with the instance instead and never save it. Or I am missing something?

Reply:
You are right but making it without the actual temporary saving to DB would require a deeper rework of the whole `save` machinery and query builder as it dives somehow to created that temporary tracker to later use it in the query builder (not sure if elsewhere) and I did not want to touch it more than necessary :smile: 

As it creates and then removes it in a transaction for the rest of the world it will look like it never happened.

Reply:
Agreed!